tests/test_gearman.py

```diff
@@ -1,29 +1,36 @@
-import unittest
-from nose.plugins.attrib import attr
-from tests.common import load_check
+# Agent
 from checks import AgentCheck
+from tests.common import AgentCheckTest, load_check
+
+# 3rd party
+from nose.plugins.attrib import attr
+
 
 @attr(requires='gearman')
-class GearmanTestCase(unittest.TestCase):
+class GearmanTestCase(AgentCheckTest):
+    CHECK_NAME = "gearmand"
+
+    
 
     def test_metrics(self):
+        tags = ['first_tag', 'second_tag']
+        service_checks_tags = ['server:127.0.0.1', 'port:4730']
         config = {
             'instances': [{
-                'tags': ['first_tag', 'second_tag']
+                'tags': tags
             }]
         }
-        agentConfig = {
-            'version': '0.1',
-            'api_key': 'toto'
-        }
-
-        self.check = load_check('gearmand', config, agentConfig)
-        self.check.check(config['instances'][0])
-
-        metrics = self.check.get_metrics()
-        self.assertTrue(type(metrics) == type([]), metrics)
-        self.assertTrue(len(metrics) == 4)
-        self.assertTrue(len([k for k in metrics if "second_tag" in k[3]['tags']]) == 4)
+        tags += service_checks_tags
+        self.run_check(config)
+        self.assertMetric('gearman.unique_tasks', value=0.0, tags=tags)
+        self.assertMetric('gearman.running', value=0.0, tags=tags)
+        self.assertMetric('gearman.queued', value=0.0, tags=tags)
+        self.assertMetric('gearman.workers', value=0.0, tags=tags)
```

Comment:
These metrics should only have one datapoint. Can you ensure this by using `count=1` in the assert?